File: problems/167/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int lo = 0, hi = size(nums) - 1;
        
        while (nums[lo] + nums[hi] != target) {
            if (nums[lo] + nums[hi] < target) {
                lo++;
            } else {
                hi--;
            }
        }
        
        return {lo + 1, hi + 1};
    }
};

File: problems/167/solution.go
package main
func twoSum(numbers []int, target int) []int {
    i, j := 0, len(numbers) - 1
    sum := numbers[i] + numbers[j]
    
    for sum != target {
        if sum < target {
            i++
        } else {
            j--
        }
        
        sum = numbers[i] + numbers[j]
    }
    
    return []int {i + 1, j + 1}
}